### ISSUE TITLE
Syntax highlighting has very limited support

### ISSUE THREAD

**Issue by noraj** [retypeapp/retype#446]
The library use as syntax highlighter is not documented in retype (see #445) so I don't know what are supported languages but I can already say it seems the library used is quite poor, I doesn't know the very famous ERB templating language that is included in any major syntax highlighter :

- [highlight.js](https://highlightjs.org/static/demo/)
- [Prism.js](https://prismjs.com/#supported-languages)
- [Rouge](http://rouge.jneen.net/v4.0.0/erb/)
- [Pygments](https://pygments.org/languages/)

Is that because all modules/languages are not loaded by default in Retype? Or because the library used has very limited languages support? If option 1 could you enable all modules, if option 2 is that possible to change of library?

**Comment by geoffreymcgill** [comment]
Retype uses [PrismJS](https://prismjs.com/) for syntax highlighting. 

It appears ERB is not well supported by Prism. I ran a test using their playground and there was basically no syntax highlighting, see:

https://prismjs.com/test.html#language=ruby

Depending on your code sample, switching to `ruby` does highlight the syntax, although I'm unsure if that creates the expected syntax highlighting result for ERB code. 

Interestingly, VS Code does not appear to syntax highlight `erb` blocks within `.md` files either (at least for me). Again, switching to `ruby` does highlight.

With Retype, all PrismJS language modules are dynamically loaded as required. All Prism [supported languages](https://prismjs.com/#supported-languages) are supported by Retype.

**Comment by noraj** [comment]
> It appears ERB is not well supported by Prism. I ran a test using their playground and there was basically no syntax highlighting, see:
> 
> https://prismjs.com/test.html#language=ruby

I don't agree, it works well, see at https://prismjs.com/test.html#language=erb:

With the follwoing template:

```erb
  <html>
    <head><title>Ruby Toys -- <%= @name %></title></head>
    <body>

      <h1><%= @name %> (<%= @code %>)</h1>
      <p><%= @desc %></p>

      <ul>
        <% @features.each do |f| %>
          <li><b><%= f %></b></li>
        <% end %>
      </ul>

      <p>
        <% if @cost < 10 %>
          <b>Only <%= @cost %>!!!</b>
        <% else %>
           Call for a price, today!
        <% end %>
      </p>

    </body>
  </html>
```

![image](https://user-images.githubusercontent.com/16578570/203093459-730c42d0-104c-47df-9176-f42759c60693.png)

> Depending on your code sample, switching to `ruby` does highlight the syntax, although I'm unsure if that creates the expected syntax highlighting result for ERB code.

No, ERB highlight both HTML and Ruby while Ruby highlight only Ruby.

> Interestingly, VS Code does not appear to syntax highlight `erb` blocks within `.md` files either (at least for me). Again, switching to `ruby` does highlight.

Yeah, VSCode highlight a very limited amount of languages, that's why there are tons of `XYZ language` and syntax highlight plugins for VSCode.

> With Retype, all PrismJS language modules are dynamically loaded as required. All Prism [supported languages](https://prismjs.com/#supported-languages) are supported by Retype.

On Prism.js website I can highlight the chunk of code I wanted with ERB:

![image](https://user-images.githubusercontent.com/16578570/203095505-393824da-e34a-46c8-ad23-24e06a299c3e.png)

While on Retype it doesn't highlight anything.

![image](https://user-images.githubusercontent.com/16578570/203095743-c3435804-5315-4dea-ae0e-4ac64ea90886.png)

The example is live [here](https://trove.raw.pm/tools/msfconsole/#set-http-headers) and you can check the source code [here](https://gitlab.com/rawsec/the-hacking-trove/-/blob/master/docs/Tools/msfconsole.md?plain=1).

**Comment by noraj** [comment]
In #445 you said

> All PrismJS language modules are dynamically loaded **as required**

So that must be it, Retype fails to load the ERB module (and maybe others) even if a code block is using the required language.

**Comment by geoffreymcgill** [comment]
We'll have to investigate. Could be the theme does not support those tokens. In the next Retype release, all the libraries are getting updated to their latest releases as well, so that might help.

**Comment by geoffreymcgill** [comment]
We found a defect in how the Prism themes were rendering and after fixing that issue, the `erb` rendering improved. At least the `red` font-color is gone now.

Latest rendering of the `erb` sample now looks like the following:

<img width="877" alt="Screen Shot 2023-04-25 at 8 30 39 PM" src="https://user-images.githubusercontent.com/62210/234451485-a642129b-62b0-45d2-aa34-571040ea39be.png">

The fix will be included in Retype v3.0.

**Comment by geoffreymcgill** [comment]
Retype v3.0 has been released and includes this fix. 

https://retype.com/guides/getting-started/#update
